node-bootstrapper/app.go

```diff
@@ -100,6 +104,60 @@ func (a *App) provisionStart(ctx context.Context, cse utils.SensitiveString) err
 	return err
 }
 
+// usage example:
+// node-bootstrapper monitor
+func (a *App) Monitor(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 15*time.Minute)
+	defer cancel()
+
+	for {
+		select {
+		case <-timeoutCtx.Done():
+			// If the timeout or cancel occurs, exit with a timeout error
+			return fmt.Errorf("monitoring timed out: %s still active after 15 minutes", BootstrapService)
+		default:
+			_, err := os.Stat(ProvisionJSONFilePath)
+			// Check the active state of the unit
+			_, sysctlerr := a.runSystemctlCommand(ctx, "status", BootstrapService)
+
+			// if service is inactive or failed, error code will be non-zero
+			if sysctlerr == nil || os.IsNotExist(err) {
+				// Unit is still active, sleep for 3 seconds before checking again
+				time.Sleep(3 * time.Second)
```

Comment:
might also want to make the sleep time configurable with a sane default

Reply:
just a random number I chose, we can tweak it if there are better suggestions. Though idk if we should increase it since you mentioned that in most cases provisioning should be complete by the time cse runs.

Reply:
On average, `3*time.Second / 2` will be added to the VM provisioning time.

Smaller number => faster VM provision. I think it should be no more than a second. Maybe less then that.

I wonder if there is a better way to watch for a file appearance.

Reply:
found a better way using `fsnotify`